geode-core/src/test/java/org/apache/geode/internal/cache/ClientServerTransactionDUnitTest.java

```diff
@@ -4065,4 +4065,157 @@ private void doJTATx2(String regionName, CountDownLatch latch1, CountDownLatch l
       Assert.fail("Unexpected exception while doing JTA Transaction2 ", e);
     }
   }
+
+  @Test
+  public void testPartitionMessageSetsClientMemberIdAsTxMemberId() {
+    Host host = Host.getHost(0);
+    VM server1 = host.getVM(0);
+    VM client = host.getVM(2);
+    int totalBuckets = 50;
+    String regionName = "region";
+
+    setupRegionForClientTransactions(totalBuckets, regionName, false, null);
+
+    client.invokeAsync(() -> doKeySetOpTransaction(1, regionName, totalBuckets, false, null));
+
+    // Should cause TXId(server1, 1) to be executed on server2
+    server1.invoke(() -> doPutOpTransaction(regionName, totalBuckets));
```

Comment:
There is no assertion in doPutOpTransaction(...): what is validating this final action? Is there an exception that would be thrown if the wrong thing happens?

Reply:
No, the test would hang without the fix. This transaction is to be conflict with (same TXId) the other transaction.